pkg/nsm/apis/internsm/internsm.proto

```diff
@@ -0,0 +1,105 @@
+service InterNsm {
+    // RequestRemoteConnection - Service called by one NSM to request a RemoteConnection to a Network Service Endpoint from another NSM
+    rpc RequestRemoteConnection (RemoteConnectionRequest) returns (RemoteConnectionReply);
+    // DeleteRemoteConnection - Service called by one NSM to to request a RemoteConnection be deleted
+    rpc DeleteRemoteConnection (RemoteConnection) returns (RemoteConnectionReply);
+    // UpdateRemoteConnection - Service called by one NSM to request update of a RemoteConnection by another NSM.
+    // Note: Only intended to Update ConnectionContext
+    rpc UpdateRemoteConnection (RemoteConnection) returns (RemoteConnectionReply);
+    // MonitorRemoteConnections - Service called by one NSM to request streaming updates on existing RemoteConnections
+    //     RemoteConnectionId - interpreted as a 'selector' with empty fields being wildcarded.
+    //                          Example: specifying source_nsm_id will cause streaming of updates for all connections matching source nsm_id
+    rpc MonitorRemoteConnections (RemoteConnectionId) returns (stream RemoteConnectionUpdate);
+}
+
+// RemoteConnectionId - uniquely identifies the connection
+// Note: A connection in Requested state may have a partially specified RemoteConnectionId.  A RemoteConnectionId must be fully specified in CONNECTED state
+message RemoteConnectionId {
+    // source_nsm_id - Id of the source NSM - provided authoritatively by source NSM in RemoteConnectionRequest
+    // Can be used for 'recovery' in the event of an NSM restart by 'Monitoring' all RemoteConnections
+    // from its peers matching its own source_nsm_id
+    string source_nsm_id = 1;
+    // destination_nsm_id - Id of the destination NSM - provided authoritatively by the destination NSM in RemoteConnectionReply
+    string destination_nsm_id = 2;
+    // source_connection_id - Id of the source NSM's end of the connection - provided authoritatively by the source NSM in RemoteConnectionRequest
+    string source_connection_id = 3;
+    // destination_connection_id - Id of the destination NSM's end of the connection - provided authoritatively by the destination NSM in the RemoteConnectionReply
+    string destination_connection_id = 4;
+}
+
+// RemoteConnectionState - state of the RemoteConnection
+enum RemoteConnectionState {
+    // REQUESTED - State of RemoteConnection in RemoteConnectionRequest
+    REQUESTED = 0;
+    // CONNECTED - State of RemoteConnection in RemoteConnectionReply in response to RequestRemoteConnection
+    CONNECTED = 1;
+    // CLOSE_REQUESTED - State of RemoteConnection passed to RemoteConnectionReply in response to DeleteRemoeConnection
+    CLOSE_REQUESTED = 2;
+    // CLOSED - State of the RemoteConnection in RemoteConnectionReply in response to DeleteRemoteConnection
+    CLOSED = 3;
+}
+
+// RemoteConnection - represents a RemoteConnection, may be partially or fully specified
+message RemoteConnection {
```

Comment:
It would be great to see why and how these fields are used.

Reply:
Good point, will do :)